internal/rbd/replication.go

```diff
@@ -19,31 +19,31 @@ package rbd
 import (
 	"context"
 	"strings"
+	"fmt"
 
 	librbd "github.com/ceph/go-ceph/rbd"
 	"github.com/csi-addons/spec/lib/go/replication"
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
 )
 
 func (rv *rbdVolume) ResyncVol(localStatus librbd.SiteMirrorImageStatus, force bool) error {
 	if resyncRequired(localStatus) {
 		// If the force option is not set return the error message to retry
 		// with Force option.
 		if !force {
-			return status.Errorf(codes.FailedPrecondition,
-				"image is in %q state, description (%s). Force resync to recover volume",
-				localStatus.State, localStatus.Description)
+		 	fmt.Errorf("image is in %q state, description (%s). Force resync to recover volume", localStatus.State, localStatus.Description)
+			return ErrFailedPrecondition
 		}
 		err := rv.resyncImage()
 		if err != nil {
-			return status.Error(codes.Internal, err.Error())
+			fmt.Errorf("failed to resync image")
+			return err
```

Comment:
What type of error is returned here? Maybe it makes sense to provide a comment at the top of the function that describes the different errors? The caller of the function can then easily figure out what errors need to be caught and handled differently.